src/main/scala/bazeltools/bsp4bazel/BazelBspServer.scala

```diff
@@ -352,11 +361,19 @@ object Bsp4BazelServer:
       }
 
   object TargetSourceMap:
-    def apply(targets: List[BspTaskRunner.BspTarget]): TargetSourceMap =
+    def apply(
+        workspaceRoot: Path,
+        targets: List[BspTaskRunner.BspTarget]
+    ): TargetSourceMap =
       TargetSourceMap(
         targets
           .map(t =>
```

Comment:
nit: I would add `targets.iterator.map(t =>` so you don't allocate the intermediate List.